copilot-chat-copilot.el

```diff
@@ -88,145 +92,149 @@
         (insert-file-contents token-file)
         (buffer-substring-no-properties (point-min) (point-max))))))
 
-
-(defun copilot-chat--create ()
-  "Create a new Copilot chat instance."
-  (setq copilot-chat--instance (copilot-chat--make
-                                :ready t
-                                :github-token (copilot-chat--get-cached-token)
-                                :token nil
-                                :sessionid (concat (copilot-chat--uuid) (number-to-string (* (round (float-time (current-time))) 1000)))
-                                :machineid (copilot-chat--machine-id)
-                                :history nil
-                                :buffers nil
-                                :last-models-fetch-time 0))
-
-  ;; Load models from cache if available
-  (let ((cached-models (copilot-chat--load-models-from-cache)))
-    (when cached-models
-      (setf (copilot-chat-models copilot-chat--instance) cached-models)
-      (message "Loaded models from cache. %d models available." (length cached-models))))
-
-  ;; Schedule background model fetching with slight delay
-  (run-with-timer 2 nil #'copilot-chat--fetch-models-async)
-
-  copilot-chat--instance)
+(defun copilot-chat--create (directory)
+  "Create a new Copilot chat instance with DIRECTORY as source directory."
+  (copilot-chat--make
+    :directory directory
+    :model copilot-chat-default-model
+    :chat-buffer nil
+    :first-word-answer t
+    :history nil
+    :buffers nil
+    :prompt-history nil
+    :prompt-history-position nil
+    :yank-index 1
+    :last-yank-start nil
+    :last-yank-end nil
+    :spinner-timer nil
+    :spinner-index 0
+    :spinner-status nil
+    :curl-answer nil
+    :curl-file nil
+    :curl-current-data nil))
 
 (defun copilot-chat--fetch-models-async ()
   "Fetch models asynchronously in the background."
   (when (copilot-chat--ready-p)
```

Comment:
Either the merging operation has failed or the function has disappeared.

Reply:
I was not sure if it is useful to fetch models at each chat creation (now we don't create only one chat anymore). Finally, I restored old function because it does not seems to be expensive.